src/app/activity/bounty/Claim.ts

```diff
@@ -10,13 +10,35 @@ import { BountyStatus } from '../../constants/bountyStatus';
 import BountyUtils from '../../utils/BountyUtils';
 import { Activities } from '../../constants/activities';
 import { Clients } from '../../constants/clients';
+import { CommandContext } from 'slash-create';
+import ValidationError from '../../errors/ValidationError';
+import WalletUtils from '../../utils/WalletUtils';
+import { UpsertUserWalletRequest } from '../../requests/UpsertUserWalletRequest';
+import { Request } from '../../requests/Request';
+import { handler } from './Handler';
+import { UserCollection } from '../../types/user/UserCollection';
 
 export const claimBounty = async (request: ClaimRequest): Promise<any> => {
     Log.debug('In Claim activity');
 
     const claimedByUser = await DiscordUtils.getGuildMemberFromUserId(request.userId, request.guildId);
     Log.info(`${request.bountyId} bounty claimed by ${claimedByUser.user.tag}`);
+
+    if (! (await isUserWalletRegistered(request))) {
+        if (request.commandContext) {
+            const gotoDMMessage = 'Go to your DMs to finish claiming the bounty...';
+            await request.commandContext.send({ content: gotoDMMessage, ephemeral: true});
+        }
+    
+        const claimWalletMessage = `Hello <@${request.userId}>!\n` +
+            `Please respond within 2 minutes.\n` +
+            `Please enter the ethereum wallet address (non-ENS) to receive the reward amount for this bounty`;
+        const walletNeededMessage: Message = await claimedByUser.send({ content: claimWalletMessage });
+        const dmChannel: DMChannel = await walletNeededMessage.channel.fetch() as DMChannel;
     
+        await BountyUtils.userInputWalletAddress(dmChannel, request.userId);
+    }
```

Comment:
After 3 bad tries at entering the address, it gave me an error message, but still claimed the bounty for me. I think you need to throw an error or detect the failure manually.

Reply:
Agh. As I was writing this I was waffling on whether it should be a requirement or just the first part of the flow, and this reflects that. Addressed in next revision.